src/three/plugins/loaders/WMSCapabilitiesLoader.js

```diff
@@ -0,0 +1,360 @@
+
+		queryable = el.getAttribute( 'queryable' ) === '1';
+
+	}
+
+	if ( el.hasAttribute( 'opaque' ) ) {
+
+		opaque = el.getAttribute( 'opaque' ) === '1';
+
+	}
+
+	if ( el.querySelector( 'EX_GeographicBoundingBox' ) ) {
+
+		exGeographicBoundingBox = parseEXGeographicBoundingBox( el.querySelector( 'EX_GeographicBoundingBox' ) );
+
+	}
+
+	// Recursively parse sublayers
+	const subLayers = Array.from( el.querySelectorAll( ':scope > Layer' ) ).map( el => {
+
+		return parseLayer( el, {
+			// add
+			styles,
+			crs,
+
+			// replace
+			exGeographicBoundingBox,
+			queryable,
+			opaque,
+		} );
+
+	} );
+
+	// TODO:
+	// - Dimension
+	// - Attribution
+	// - AuthorityURL
+	// - MinScaleDenominator
+	// - MaxScaleDenominator
+	// - cascaded, noSubsets, fixedWidth, fixedHeight attributes
+
+	return {
+		name,
+		title,
+		abstract,
+		queryable,
+		opaque,
+		keywords,
+		crs,
+		boundingBoxes,
+		exGeographicBoundingBox,
+		styles,
+		subLayers,
+	};
+
+}
+
+function parseService( el ) {
+
+	return {
+		name: el.querySelector( 'Name' )?.textContent || '',
+		title: el.querySelector( 'Title' )?.textContent || '',
+		abstract: el.querySelector( 'Abstract' )?.textContent || '',
+		keywords: Array.from( el.querySelectorAll( 'Keyword' ) ).map( k => k.textContent ),
+
+		maxWidth: parseFloat( el.querySelector( 'MaxWidth' ) ) || null,
+		maxHeight: parseFloat( el.querySelector( 'MaxHeight' ) ) || null,
+		layerLimit: parseFloat( el.querySelector( 'LayerLimit' ) ) || null,
+	};
+
+}
+
+// helper: read OnlineResource href (handles xlink namespace)
+function readOnlineResourceHref( el ) {
+
+	if ( ! el ) return '';
+	return (
+		el.getAttribute( 'xlink:href' ) ||
+		el.getAttributeNS( 'http://www.w3.org/1999/xlink', 'href' ) ||
+		''
+	).trim();
+
+}
+
+// parse a single operation (e.g. GetMap, GetCapabilities, GetFeatureInfo)
+function parseRequestOperation( opEl ) {
+
+	const formats = Array.from( opEl.querySelectorAll( 'Format' ) ).map( f => f.textContent.trim() );
+	const dcp = Array.from( opEl.querySelectorAll( 'DCPType' ) ).map( dcp => {
+
+		const httpEl = dcp.querySelector( 'HTTP' );
+		if ( ! httpEl ) {
+
+			return { type: 'UNKNOWN', get: '', post: '' };
+
+		}
+
+		const getEl = httpEl.querySelector( 'Get OnlineResource' ) || httpEl.querySelector( 'Get > OnlineResource' ) || httpEl.querySelector( 'Get' );
+		const postEl = httpEl.querySelector( 'Post OnlineResource' ) || httpEl.querySelector( 'Post > OnlineResource' ) || httpEl.querySelector( 'Post' );
+		const getHref = readOnlineResourceHref( getEl );
+		const postHref = readOnlineResourceHref( postEl );
+		return { type: 'HTTP', get: getHref, post: postHref };
+
+	} );
+
+	// fallback: sometimes OnlineResource appears directly under the operation
+	if ( dcp.length === 0 ) {
+
+		const online = opEl.querySelector( ':scope > OnlineResource' );
+		if ( online ) {
+
+			dcp.push( { type: 'HTTP', get: readOnlineResourceHref( online ), post: '' } );
+
+		}
+
+	}
```

Comment:
For these lines here - do you have examples of some of these non-standard ways of representing the "OnlineResource" fields? According to [this scheme](https://schemas.opengis.net/wms/1.3.0/capabilities_1_3_0.xsd) the "OnlineResource" field is supposed to _always_ be inside a "Get" and "Post" element. But it looks like you're checking for the url in a few different places. My preference is to only check for the expected location as defined by the specification but if its common for some of the services to place it in odd locations then we can keep the check here - I just want to make sure it's needed.